File: server/src/Services/UserService/UserService.Domain/Entities/Brand.cs
namespace UserService.Domain.Entities;

public class Brand
{
    public Guid UserId { get; set; }
    public User? User { get; set; }
    public string? Name { get; set; } = string.Empty;
    public double? Latitude { get; set; }
    public double? Longitude { get; set; }
    public string? Address { get; set; } = string.Empty;
    public string? Domain { get; set; } = string.Empty;
}

File: client/mobile/vousphere/lib/data/models/Brand.dart
class Brand {
  String brandId;
  String name;
  double? latitude;
  double? longitude;
  String address;
  String domain;

  Brand({
    required this.brandId,
    required this.name,
    this.latitude,
    this.longitude,
    required this.address,
    required this.domain,
  });

  factory Brand.fromJson(Map<String, dynamic> json) {
    return Brand(
      brandId: json['brandId'] ?? '',
      name: json['name'] ?? '',
      latitude: json['latitude'] != null
          ? (json['latitude'] as num).toDouble()
          : null,
      longitude: json['longitude'] != null
          ? (json['longitude'] as num).toDouble()
          : null,
      address: json['address'] ?? '',
      domain: json['domain'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'brandId': brandId,
      'name': name,
      'latitude': latitude,
      'longitude': longitude,
      'address': address,
      'domain': domain,
    };
  }

  @override
  String toString() {
    return toJson().toString();
  }
}
